.github/workflows/on-pr-debug.yml

```diff
@@ -77,10 +77,83 @@ jobs:
         env:
           VCPKG_BINARY_SOURCES: "clear;x-gha,readwrite"
         run: |
-          cmake -B build -S . -DCMAKE_C_FLAGS=/W4 -DNATS_BUILD_STREAMING=OFF
+          cmake -B build -S . -DCMAKE_C_FLAGS=/W4
           cmake --build build --config Debug
 
       - name: Test
+        shell: bash
         run: |
           cd build
+          
           ./bin/Debug/testsuite
+          mv list.txt ../test/list.txt
+          
+          # refresh the cache
+          cmake -S .. -B .
+          cmake --build . --config Debug
+
+          # Download latest nats-server
+          rel="latest" # TODO: parameterize
+          if [ "$rel" = "latest" ]; then
+            rel=$(curl -s https://api.github.com/repos/nats-io/nats-server/releases/latest | jq -r '.tag_name')
+          fi
+          if [ "$rel" != "${rel#v}" ] && wget https://github.com/nats-io/nats-server/releases/download/$rel/nats-server-$rel-windows-amd64.tar.gz; then
+            tar -xzf nats-server-$rel-linux-amd64.tar.gz
+            cp nats-server-$rel-windows-amd64/nats-server.exe ../deps/nats-server/nats-server.exe
+          else
+            for c in 1 2 3 4 5
+            do
+              echo "Attempt $c to download binary for main"
+              rm -f ./nats-server
+              curl -sf "https://binaries.nats.dev/nats-io/nats-server/v2@$rel" | PREFIX=. sh
+              # We are sometimes getting nats-server of size 0. Make sure we have a
+              # working nats-server by making sure we get a version number.
+              mv ./nats-server ./nats-server.exe
+              v="$(./nats-server.exe -v)"
+              if [ "$v" != "" ]; then
+                break
+              fi
+            done
+            mkdir -p ../deps/nats-server
+            mv ./nats-server.exe ../deps/nats-server/nats-server.exe
+          fi
+          
+          export PATH=../deps/nats-server:$PATH
+          export NATS_TEST_SERVER_VERSION="$(nats-server -v)"
+
+          run_ctest() {
+            local extra_args="$@"
+            local current_set=$(set +o)
+            set +e
+            echo "Running ctest($extra_args)..."
+
+            ctest -C Debug --timeout 60 --output-on-failure $extra_args
+            
+            local result=$?
+            eval "$current_set"
+            return $result
+          }
+
+          run_ctest
+          result=$?
+      
+          # Retry if failed
```

Comment:
I am 0/5 against auto-retries. They mask out the (annoying) flappers, but I'd rather be annoyed and more motivated to fix them, TBH.

Reply:
unfortunately there aren't a definitive set of failing tests but there seems to be an issue with 'something' however much I excluded failing tests, one or two always fails! (I'm guessing nats-server process management, just a guess, btw having similar issues with nats.net tests on windows).

if we removed retires tests almost certain to never pass and hide potential real failures which might be useful. having said that, i see your point as well.

if you still think the same please let me know i can remove it.

Reply:
Since this logic applies to Windows only, I'm good with it for now, but would be great to get to the bottom of this issue. There are 2-3 "known" flappers at the moment, but the rest - 2/5 we should aim to pass, maybe with specific exclusions as applicable.